src/main/java/be/seeseemelk/mockbukkit/WorldBorderMock.java

```diff
@@ -0,0 +1,188 @@
+package be.seeseemelk.mockbukkit;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Location;
+import org.bukkit.Server;
+import org.bukkit.World;
+import org.bukkit.WorldBorder;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.util.BoundingBox;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A mock world border object.
+ */
+public class WorldBorderMock implements WorldBorder
+{
+
+	private static final double DEFAULT_BORDER_SIZE = 6.0E7;
+	private static final double DEFAULT_DAMAGE_AMOUNT = 0.2;
+	private static final double DEFAULT_DAMAGE_BUFFER = 5.0;
+	private static final int DEFAULT_WARNING_DISTANCE = 5;
+	private static final int DEFAULT_WARNING_TIME = 15;
+	private static final double DEFAULT_CENTER_X = 0;
+	private static final double DEFAULT_CENTER_Z = 0;
+	private static final double MAX_CENTER_VALUE = 3.0E7D;
+
+	private final World world;
+	private final Server server;
+	private double size;
+	private double damageAmount;
+	private double damageBuffer;
+	private int warningDistance;
+	private int warningTime;
+	private double centerX;
+	private double centerZ;
+
+	/**
+	 * Creates a new world border mock
+	 *
+	 * @param world The world it is the border of
+	 * @param server The server it is in
+	 */
+	public WorldBorderMock(World world, Server server)
+	{
+		this.world = world;
+		this.server = server;
+
+		reset();
+	}
+
+	@Override
+	public void reset()
+	{
+		setSize(DEFAULT_BORDER_SIZE);
+		setDamageAmount(DEFAULT_DAMAGE_AMOUNT);
+		setDamageBuffer(DEFAULT_DAMAGE_BUFFER);
+		setWarningDistance(DEFAULT_WARNING_DISTANCE);
+		setWarningTime(DEFAULT_WARNING_TIME);
+		setCenter(DEFAULT_CENTER_X, DEFAULT_CENTER_Z);
+	}
+
+	@Override
+	public double getSize()
+	{
+		return size;
+	}
+
+	@Override
+	public void setSize(double newSize)
+	{
+		this.size = newSize;
+	}
+
+	@Override
+	public void setSize(double newSize, long seconds)
+	{
+		if (seconds > 0)
+		{
+			// Assumes server at perfect 20tps
+			double distance = newSize - size;
+			double ticksToTake = seconds * 20;
+
+			double distancePerTick = distance / ticksToTake;
+
+			server.getScheduler().runTaskTimer(null, new BukkitRunnable()
+			{
+				@Override
+				public void run()
+				{
+					if ((size < newSize && distance > 0) || (size > newSize && distance < 0))
+					{
+						size += distancePerTick;
+					}
+					else
+					{
+						size = newSize;
+						this.cancel();
+					}
+				}
+			}, 1, 1);
```

Comment:
We can use a lambda-expression here.
Also is `null` valid to use here?

```suggestion
			server.getScheduler().runTaskTimer(null, () ->
			{
				if ((size < newSize && distance > 0) || (size > newSize && distance < 0))
				{
					size += distancePerTick;
				}
				else
				{
					size = newSize;
					this.cancel();
				}
			}, 1, 1);
```

Reply:
> Also is `null` valid to use here?

Unless there is an system for internally scheduling repeating tasks inside mockbukkit, then I think that this is the only way to do it. The null should not be a problem as all plugin is used for is canceling tasks that were created by the plugin in `BukkitSchedulerMock#cancelTasks`.

Reply:
> We can use a lambda-expression here.
It appears that a lambada cannot be used as the cancel method is called.